lib/matplotlib/axes/_base.py

```diff
@@ -2379,7 +2369,7 @@ def draw(self, renderer=None, inframe=False):
                 self.patch.get_path(),
                 self.patch.get_transform()))
 
-            renderer.draw_image(gc, round(l), round(b), im)
+            renderer.draw_image(gc, round(l), round(b), data)
```

Comment:
It ends up at the zorder of the first image in the list.  You can see `Figure.draw` or `Axes.draw` for where that actually happens.  It's not perfect (and in fact is sort of a long-standing bug in zorder), but zorder is a mess in general in matplotlib, I don't think we'll fix it at this juncture ;)